### PR TITLE
build: Place CILIUM_ENVOY_SHA dependency in the Dockerfile

### PR DESCRIPTION
**Pull request by jrajahalme**
Automatically triggered builds (such as cilium/cilium:latest) can't
run "make" to start the docker build, but must build from the
Dockerfile without any prepatory steps. Enable this by moving the
CILIUM_ENVOY_SHA dependency to the Dockerfile instead of a separate
file. Makefile.defs still defines CILIUM_ENVOY_SHA as a variable, but
gets it from the Dockerfile.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6158)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by jrajahalme** [comment]
Unrelated CI flake, testing again

**Comment by jrajahalme** [comment]
test-me-please

File: pkg/envoy/envoy.go [hunk 1]
```diff
@@ -39,10 +39,9 @@ import (
 var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "envoy-manager")
 
 var (
-	// Running Envoy version will be checked against `RequiredEnvoyVersionSHA`.
-	// By default cilium-agent will fail to start if there is a version mismatch.
-	// This value must match the one in file CILIUM_ENVOY_SHA.
-	RequiredEnvoyVersionSHA = "fd3866831d8f85e27120f34d98133343fc9c4e0e"
+	// RequiredEnvoyVersionSHA is set during build
+	RequiredEnvoyVersionSHA string
```
Comment (ianvernon) [inline pkg/envoy/envoy.go:45]: Can you keep the log message about how `cilium-agent` will fail to start if there is a version mismatch?

Comment (jrajahalme) [reply pkg/envoy/envoy.go:45]: Done

**Comment by jrajahalme** [comment]
Hit by a new issue fixed in #6160, which is also included here to help pass the CI.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
Unrelated CI issues, testing again.

**Comment by jrajahalme** [comment]
test-me-please

previous failure https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/7864/testReport/junit/k8s-1/12/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/

**Comment by aanm** [comment]
test-me-please

**Pull request merged**